### PR TITLE
Added warnings to deprecated arguments

### LINKED ISSUE #3603: Deprecations in SeqFeature
Is there a schedule to add the deprecation warning as per:

https://github.com/biopython/biopython/blob/870f623005ed1b95df877f618cbb12a815bd5696/Bio/SeqFeature.py#L158-L175

### PR DESCRIPTION
**Pull request by timjden**
- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``pre-commit`` locally,
and understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

Closes #3603. I added the warnings for the deprecated arguments and edited the docstring. Hope it's OK - this is my first PR ever.

### CODE REVIEW

**Comment by timjden** [comment]
Thanks. I will look into the testing. For now I have added "in Release 1.80" to `DEPRECATION.rst`.

**Comment by timjden** [comment]
@peterjc I am unsure why the build would fail on CircleCI after that tiny change (5c7a376). What is your advice here? Many thanks.

File: DEPRECATED.rst [hunk 1]
```diff
@@ -267,6 +267,13 @@ Bio.SeqFeature
 With the introduction of the CompoundLocation in Release 1.62, the SeqFeature
 attribute sub_features was deprecated. It was removed in Release 1.68.
 
+Note that location_operator argument can no longer
+be used, instead do this via the CompoundLocation object.
+
+Note that the strand, ref and ref_db arguments to the SeqFeature are
+now deprecated and will later be removed. Set them via the location
+object instead.
+
```
[the remark this answers is not in the document]
Comment (peterjc) [reply DEPRECATED.rst:276]: CircleCI is running our pre-commit checks, which are saying:

```
DEPRECATED.rst:274: D001 Line too long
```

It is right, the line "now deprecated in Release 1.80 and will later be removed. Set them via the location" is 83 characters long. Try to line wrap at 78 characters ideally - both new paragraphs.

See https://github.com/biopython/biopython/blob/master/CONTRIBUTING.rst for how to run these checks on your own machine.

[the remark this answers is not in the document]
Comment (peterjc) [reply DEPRECATED.rst:276]: I have fixed this now.

**Comment by peterjc** [comment]
This *should* pass the automated tests now, after which I intend to do a squash-and-merge commit.

**Comment by peterjc** [comment]
That's merged now, thank you 🎉

**Pull request merged**